packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,246 @@
+        _infile;
+        Input memory parsedInput; // TODO parse the input file.
+        loadInput(parsedInput);
+        require(false, "DeploySuperchainInput: loadInput is not implemented");
+    }
+
+    function loadInput(Input memory _input) public {
+        require(!inputSet, "DeploySuperchainInput: input already set");
+        input = _input;
+    }
+
+    // -------- Getter methods for inputs --------
+
+    function inputs() public view returns (Input memory) {
+        require(inputSet, "DeploySuperchainInput: input not set");
+        return input;
+    }
+
+    function proxyAdminOwner() public view returns (address out) {
+        out = input.roles.proxyAdminOwner;
+        require(out != address(0), "DeploySuperchainInput: proxyAdminOwner not set");
+    }
+
+    function protocolVersionsOwner() public view returns (address out) {
+        out = input.roles.protocolVersionsOwner;
+        require(out != address(0), "DeploySuperchainInput: protocolVersionsOwner not set");
+    }
+
+    function guardian() public view returns (address out) {
+        out = input.roles.guardian;
+        require(out != address(0), "DeploySuperchainInput: guardian not set");
+    }
+
+    function paused() public view returns (bool out) {
+        out = input.paused;
+    }
+
+    function requiredProtocolVersion() public view returns (ProtocolVersion out) {
+        out = input.requiredProtocolVersion;
+        require(ProtocolVersion.unwrap(out) != 0, "DeploySuperchainInput: requiredProtocolVersion not set");
+    }
+
+    function recommendedProtocolVersion() public view returns (ProtocolVersion out) {
+        out = input.recommendedProtocolVersion;
+        require(ProtocolVersion.unwrap(out) != 0, "DeploySuperchainInput: recommendedProtocolVersion not set");
+    }
+}
+
+contract DeploySuperchainOutput {
+    struct Output {
+        ProxyAdmin superchainProxyAdmin;
+        SuperchainConfig superchainConfigImpl;
+        SuperchainConfig superchainConfigProxy;
+        ProtocolVersions protocolVersionsImpl;
+        ProtocolVersions protocolVersionsProxy;
+    }
+
+    Output internal output;
+    bool internal outputSet = false;
+
+    // -------- Save outputs --------
+
+    function saveOutput(Output memory _output) public {
+        require(!outputSet, "DeploySuperchainOutput: output already set");
+        output = _output;
+        outputSet = true;
+    }
+
+    function writeOutput(string memory _outfile) public view {
+        require(outputSet, "DeploySuperchainOutput: output not set");
+        _outfile; // TODO: write to file.
+        require(false, "DeploySuperchainOutput: saveOutput not implemented");
+    }
+
+    // -------- Getter methods for outputs --------
+    function outputs() public view returns (Output memory) {
+        require(outputSet, "DeploySuperchainOutput: output not set");
+        return output;
+    }
+
+    function superchainProxyAdmin() public view returns (ProxyAdmin out) {
```

Comment:
Same applies to outputs here, as state-vars may be less verbose than as individual getters